### DISCUSSION TITLE
Utilizing s3prl & fairseq in ESPnet

### DISCUSSION THREAD

**Discussion by Bereket-Desbele** [espnet/espnet#5209] [Q&A]
I've been trying to utilize SSLRS & wav2vec2 model to train an ASR on a small sized low resource language using ESPnet (version 2). & Every time I try to add  the s3prl & fairseq packages as follows: 

# cd ~/espnet/tools

# for s3prl using the command 
. ./activate_python.sh && ./installers/install_s3prl.sh

# and for faiseq using the command 
. ./activate_python.sh && ./installers/install_fairseq.sh

The installation at first goes smoothly but at the end the  following error message appears saying "s3prl 0.4.2 requires omegaconf>=2.1.1 but you have omegaconf 2.0.6".

Now, I have tried it several times changing the python version of my virtual environment from 3.8 to 3.9 but the error still persists. I've also medled with different torch & cudda versions to try & see if it could get fixed but to no avail.

P.S. I've also went through the ESPnet Docs & even watched the CMU Speech recognition tutorials for Fall of 2022 lecture 6 (espnet tutorial #1), where the presenter dives into the installation process. The same error actually occurred in the tutorial but he ignored it since he wasn't going to use the  packages then.

I would really appreciate it, if anyone could help me with this??? Thanks in advance.

**Comment by sw005320** [comment]
Thanks a lot for the report.
@simpleoier, can you check this?

**Comment by simpleoier** [comment]
Hi @Bereket-Desbele. Thanks for the report. I'm sorry about this error.
I guess you can ignore the error. Omegaconf majorly affects the training / inference in fairseq. It doesn't affect loading the models probably. 
As far as I know, if you use wav2vec2 as SSLR frontend, you do not need to install fairseq. Because s3prl has removed the dependencies on fairseq or other ad-hoc repos. 
But I would like to know some details. Did you install s3prl first and then fairseq? Or the other way around?

**Comment by Bereket-Desbele** [comment]
Even though it displays the error message, it indicates the installation of both s3prl & fairseq is done with the [X] marks on the list of added pancakes. Yet I can't use pre-trained models to train with my own data. The omegaconf error still persists. I am not 100% sure but the order was fairseq & then s3prl. And currently I am using ESPnet 012023 with python 3.8 & torch 1.13 & cuda 11.3.

**Reply by simpleoier** [reply]
Can you elaborate more how you ran training? Plus some error traceback information might be more helpful.

**Comment by Bereket-Desbele** [comment]
Its okay. if you don't install fairseq then all works out.
